proxy-node-translator/src/main/java/uk/gov/ida/notification/translator/saml/HubResponseTranslator.java

```diff
@@ -43,9 +45,16 @@ public HubResponseTranslator(
     Response getTranslatedHubResponse(HubResponseContainer hubResponseContainer, CountryMetadataResponse countryMetadataResponse) {
         final List<EidasAttributeBuilder> eidasAttributeBuilders = new ArrayList<>();
 
-        final String pid = hubResponseContainer.getPid()
-                .map(p -> String.format(PID_PREFIX, countryMetadataResponse.getCountryCode(), p))
-                .orElse(null);
+        final String pid;
+        final SecureRandomIdentifierGenerationStrategy idGeneratorStrategy = new SecureRandomIdentifierGenerationStrategy();
+
+        if (hubResponseContainer.isEidasTransientPid()){
+            pid = String.format(PID_PREFIX, countryMetadataResponse.getCountryCode(), TRANSIENT_PREFIX + idGeneratorStrategy.generateIdentifier(true));
+        } else {
+            pid = hubResponseContainer.getPid()
+                    .map(p -> String.format(PID_PREFIX, countryMetadataResponse.getCountryCode(), p))
+                    .orElse(null);
+        }
```

Comment:
This can be refactored. First, get the "raw" pid value as either the value from the hub or from the generator plus the Transient prefix. Then plug that into the `String.format()`. The below code can be a function
```
final String pid;
if (hubResponseContainer.isEidasTransientPid())
   pid = TRANSIENT_PREFIX + ... 
else
   return null if pid.empty()?
   pid = ...

return String.format(...., pid, ...)
```